civihr_default_theme/assets/js/civihr_default_theme.script.js

```diff
@@ -147,6 +175,30 @@
     }
   }
 
+  function handleWebformCalendar () {
+    // Remove Required attribute from default select datepicker, which is not used
+    $('.webform-container-inline.webform-datepicker div.form-item.form-type-select select').attr('required', false);
+    // Switch between Normal and Native Datepicker based on device
+    $('.mobile .webform-calendar').remove();
+    $('body:not(.mobile) .mobile-webform-calendar').remove();
+
+    // Seeter and Getter for Native Calendar
+    $('.mobile-webform-calendar').change(function () {
+      var date = new Date(this.value);
+      $('#' + this.id + '-' + 'month').val(date.getMonth() + 1);
+      $('#' + this.id + '-' + 'day').val(date.getDate());
+      $('#' + this.id + '-' + 'year').val(date.getFullYear());
+    });
+    $('.mobile-webform-calendar').each(function () {
```

Comment:
same comment as above.

Reply:
The issue is that it took some time to under the reason for these functions. Please try to explain why.